File: bryniarskyi_nazarii/bridge/shape/impl/Rectangle.java
package org.example.bridge.shape.impl;

import org.example.bridge.color.Color;
import org.example.bridge.shape.Shape;

public class Rectangle extends Shape {

    private final double sideA;
    private final double sideB;

    public Rectangle(Color color, double sideA, double sideB) {
        super(color);
        this.sideA = sideA;
        this.sideB = sideB;
    }


    @Override
    public double getSquare() {
        return sideA * sideB;
    }

    @Override
    public double getPerimeter() {
        return 2 * (sideA + sideB);
    }
}

File: terletska_oksana/bridge/Figure/Rectangle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

public class Rectangle : Figure
{
    protected double sideA;
    protected double sideB;

    public Rectangle(Color color, double sideA, double sideB) : base(color)
    {
        if (sideA <= 0 || sideB <= 0)
        {
            throw new ArgumentException("Сторони мають бути > 0");
        }

        this.sideA = sideA;
        this.sideB = sideB;
    }

    public override string Name()
    {
        return "Rectangle";
    }

    public override double Area()
    {
        return sideA * sideB;

        //if (sideA <= 0 || sideB <= 0)
        //{
        //    throw new ArgumentException("Сторони мають бути > 0");
        //}
        //else
        //{
        //    return sideA * sideB;
        //}
    }

    public override double Perimeter()
    {
        return 2 * (sideA + sideB);

        //if (sideA <= 0 || sideB <= 0)
        //{
        //    throw new ArgumentException("Сторони мають бути > 0");
        //}
        //else
        //{
        //    return 2 * (sideA + sideB);
        //}
    }

    public void Print()
    {
        Console.WriteLine("\nName: "+ Name());
        Console.WriteLine("Perimeter: "+ Perimeter());
        Console.WriteLine("Area: "+ Area());
        Console.WriteLine("Color: "+ color.Draw());
    }
}

   


namespace Bridge.Figure
{
    internal class Rectangle
    {
    }
}
